lib/api/unittest/CDataFrameAnalyzerTest.cc

```diff
@@ -1191,3 +1189,54 @@ void CDataFrameAnalyzerTest::testRunBoostedTreeTrainingWithStateRecoverySubrouti
     assertDoublesEqual("hyperparam_feature_bag_fraction");
     assertDoublesArrayEqual("hyperparam_feature_sample_probabilities");
 }
+
+maths::CBoostedTreeFactory::TBoostedTreeUPtr
+CDataFrameAnalyzerTest::getFinalTree(const TStrVec& persistedStates,
+                                     std::unique_ptr<core::CDataFrame>& frame) const {
+    CTestDataSearcher dataSearcher(persistedStates.back());
+    auto decompressor{std::make_unique<core::CStateDecompressor>(dataSearcher)};
+    decompressor->setStateRestoreSearch(api::ML_STATE_INDEX,
+                                        api::getRegressionStateId("testJob"));
+    auto stream{decompressor->search(1, 1)};
+    return maths::CBoostedTreeFactory::constructFromString(*stream, *frame);
+}
+
+std::unique_ptr<core::CDataFrame>
+CDataFrameAnalyzerTest::passDataToAnalyzer(const TDoubleVec& weights,
```

Comment:
My feeling is if we're going to factor this out we should have a utility which allows you define the target relationship as a callback and use it in all test code which adds data for regression, i.e. something like `setupRegressionData`. As it is there are some unit tests which generate and pass data to the analyser and one that doesn't.

Reply:
Let's discuss your idea offline to avoid misunderstandings.

Reply:
Done